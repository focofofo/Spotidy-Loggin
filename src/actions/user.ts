import { UserInformationPayload } from '../models'

export const SET_ACCESS_TOKEN = 'SET_ACCESS_TOKEN'
export const SET_USER_INFORMATION = 'SET_USER_INFORMATION'
export const LOGOUT_USER = 'LOGOUT_USER'

/**
 * Set access_token to the user's state
 * @param payload access_token for the user
 */
export function doSetAccessToken(payload: string) {
  return {
    type: SET_ACCESS_TOKEN,
    payload,
  }
}

/**
 * Set the user information from API response
 * @param payload id, display_name from the user
 */
export function doSetUserInformation(payload: UserInformationPayload) {
  return {
    type: SET_USER_INFORMATION,
    payload,
  }
}

/**
 * Log out the current user
 */
export function doLogoutUser() {
  return { type: LOGOUT_USER }
}
