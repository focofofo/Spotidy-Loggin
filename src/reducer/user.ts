import { AnyAction } from 'redux'
import { UserState } from '../models'
import {
  LOGOUT_USER,
  SET_ACCESS_TOKEN,
  SET_USER_INFORMATION,
} from '../actions/user'

const initialState: UserState = {
  id: '',
  display_name: '',
  access_token: '',
  isLogged: false,
}

export default function user(
  state: UserState = initialState,
  action: AnyAction
) {
  switch (action.type) {
    case SET_ACCESS_TOKEN:
      return {
        ...state,
        access_token: action.payload,
      }
    case SET_USER_INFORMATION:
      return {
        ...state,
        id: action.payload.id,
        display_name: action.payload.display_name,
        isLogged: true,
      }
    case LOGOUT_USER:
      return {
        ...initialState,
      }
    default:
      return state
  }
}
