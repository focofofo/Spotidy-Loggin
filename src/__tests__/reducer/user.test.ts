import { AnyAction } from 'redux'
import user from '../../reducer/user'
import {
  SET_ACCESS_TOKEN,
  SET_USER_INFORMATION,
  LOGOUT_USER,
} from '../../actions/user'

describe('user reducer', () => {
  it('should handle initial state', () => {
    expect(user(undefined, {} as AnyAction)).toMatchSnapshot()
  })

  it('should handle SET_ACCESS_TOKEN', () => {
    expect(
      user(undefined, {
        type: SET_ACCESS_TOKEN,
        payload: 'Test',
      })
    ).toMatchSnapshot()
  })

  it('should handle SET_USER_INFORMATION', () => {
    expect(
      user(undefined, {
        type: SET_USER_INFORMATION,
        payload: {
          id: 'TestId',
          display_name: 'Test',
        },
      })
    ).toMatchSnapshot()
  })

  it('should handle LOGOUT_USER', () => {
    expect(user(undefined, { type: LOGOUT_USER })).toMatchSnapshot()
  })
})
