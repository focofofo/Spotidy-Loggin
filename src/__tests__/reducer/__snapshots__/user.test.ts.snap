// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user reducer should handle LOGOUT_USER 1`] = `
Object {
  "access_token": "",
  "display_name": "",
  "id": "",
  "isLogged": false,
}
`;

exports[`user reducer should handle SET_ACCESS_TOKEN 1`] = `
Object {
  "access_token": "Test",
  "display_name": "",
  "id": "",
  "isLogged": false,
}
`;

exports[`user reducer should handle SET_USER_INFORMATION 1`] = `
Object {
  "access_token": "",
  "display_name": "Test",
  "id": "TestId",
  "isLogged": true,
}
`;

exports[`user reducer should handle initial state 1`] = `
Object {
  "access_token": "",
  "display_name": "",
  "id": "",
  "isLogged": false,
}
`;
